lib/generator.js

```diff
@@ -548,19 +564,38 @@ class Generator {
     } else {
       filename = filenamify(filename, { replacement: '-', maxLength: 255 });
     }
+
     const newFileName = fileName.replace(`\$\$${template}\$\$`, filename);
     const targetFile = path.resolve(this.targetDir, relativeBaseDir, newFileName);
     const relativeTargetFile = path.relative(this.targetDir, targetFile);
-    const shouldOverwriteFile = await this.shouldOverwriteFile(relativeTargetFile);
+    const shouldOverwriteFile = this.shouldOverwriteFile(relativeTargetFile);
     if (!shouldOverwriteFile) return;
     //Ensure the same object are parsed to the renderFile method as before.
     const temp = {};
     const key = template === 'everySchema' || template === 'objectSchema' ? 'schema' : template;
     temp[`${key}Name`] = name;
     temp[key] = component;
-    const content = await this.renderFile(asyncapiDocument, path.resolve(baseDir, fileName), temp);
+    await this.renderAndWriteToFile(asyncapiDocument, path.resolve(baseDir, fileName), targetFile, temp);
+  }
 
-    await writeFile(targetFile, content, 'utf8');
+  /**
+   * Renders a template and writes the result into a file.
+   * 
+   * @private
+   * @param {AsyncAPIDocument} asyncapiDocument AsyncAPI document to pass to the template.
+   * @param {String} templateFilePath Path to the file being rendered.
+   * @param {String} outputpath Path to the file being rendered.
```

Comment:
The description of these two is the same. Maybe something like this?

```suggestion
   * @param {String} templateFilePath Path to the input file being rendered.
   * @param {String} outputPath Path to the resulting rendered file.
```

Reply:
Yea good catch 😅